app/src/main/java/edu/mit/mitmobile2/tour/adapters/TourStopRecyclerViewAdapter.java

```diff
@@ -69,5 +83,26 @@ public void onBindViewHolder(ViewHolder viewHolder, final int position) {
         final MITTourStop tourStop = tourStops.get(position % tourStops.size());
         Picasso.with(context).load(tourStop.getThumbnailImage().getUrl()).fit().centerCrop().into(viewHolder.stopImageView);
         viewHolder.stopTitleTextView.setText((tourStop.getIndex() + 1) + ". " + tourStop.getTitle());
+
+        viewHolder.recyclerViewLayout.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (tourStop.getType().equals(Constants.Tours.MAIN_LOOP)) {
+                    callback.showMainLoopFragment(getStopIndexInMainLoop(tourStop));
+                } else {
+                    callback.showSideTripFragment(tourStop);
+                }
+            }
+        });
+    }
+
+    public int getStopIndexInMainLoop(MITTourStop tourStop) {
```

Comment:
If the stop is already a main loop stop, the index is going to be the same as in tourStops (or it should be, unless something weird is happening), so this function is unnecessary